Library/Homebrew/livecheck/strategy/pypi.rb

```diff
@@ -58,40 +58,48 @@ def self.generate_input_values(url)
           match = File.basename(url).match(FILENAME_REGEX)
           return values if match.blank?
 
-          # It's not technically necessary to have the `#files` fragment at the
-          # end of the URL but it makes the debug output a bit more useful.
-          values[:url] = "https://pypi.org/project/#{T.must(match[:package_name]).gsub(/%20|_/, "-")}/#files"
-
-          # Use `\.t` instead of specific tarball extensions (e.g. .tar.gz)
-          suffix = T.must(match[:suffix]).sub(Strategy::TARBALL_EXTENSION_REGEX, ".t")
-          regex_suffix = Regexp.escape(suffix).gsub("\\-", "-")
-
-          # Example regex: `%r{href=.*?/packages.*?/example[._-]v?(\d+(?:\.\d+)*(?:[._-]post\d+)?)\.t}i`
-          regex_name = Regexp.escape(T.must(match[:package_name])).gsub(/\\[_-]/, "[_-]")
-          values[:regex] =
-            %r{href=.*?/packages.*?/#{regex_name}[._-]v?(\d+(?:\.\d+)*(?:[._-]post\d+)?)#{regex_suffix}}i
+          package_name = T.must(match[:package_name]).gsub(/[_-]/, "-")
+          values[:url] = "https://pypi.org/project/#{package_name}/#files"
+          values[:regex] = %r{href=.*?/packages.*?/#{package_name}[._-]v?(\d+(?:\.\d+)*(?:[._-]post\d+)?)\.t}i
 
           values
         end
 
-        # Generates a URL and regex (if one isn't provided) and passes them
-        # to {PageMatch.find_versions} to identify versions in the content.
+        # Fetches the latest version of the package from the PyPI JSON API.
         #
         # @param url [String] the URL of the content to check
-        # @param regex [Regexp] a regex used for matching versions in content
+        # @param regex [Regexp] a regex used for matching versions in content (optional)
```

Comment:
```suggestion
        # @param regex [Regexp] a regex used for matching versions in content
        # @param provided_content [String, nil] content to check instead of
        #   fetching
        # @param homebrew_curl [Boolean] whether to use brewed curl with the URL
```

We don't need to explicitly specify that `regex` is optional in the doc comment, as the parameter is optional in the method signature and that will be apparent in the generated documentation.

Besides that, this adds fields to align with the `Crate` approach. [I have some in-progress work to expand `provided_content` (renaming it to `content`) to almost all existing strategies, as a way of increasing test coverage for now but later I will be using it for content caching.]